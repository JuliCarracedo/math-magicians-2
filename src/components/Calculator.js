/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import GrayBtn from './GrayBtn';
import OrangeBtn from './OrangeBtn';
import BigGrayBtn from './BigGrayBtn';
import Screen from './Screen';

class Calculator extends React.Component {
  render() {
    return (
      <div className="calc-body">
        <Screen content="0" />
        <div className="row">
          <GrayBtn symbol="AC" />
          <GrayBtn symbol="+/-" />
          <GrayBtn symbol="%" />
          <OrangeBtn symbol="รท" />
        </div>
        <div className="row">
          <GrayBtn symbol="7" />
          <GrayBtn symbol="8" />
          <GrayBtn symbol="9" />
          <OrangeBtn symbol="X" />
        </div>
        <div className="row">
          <GrayBtn symbol="4" />
          <GrayBtn symbol="5" />
          <GrayBtn symbol="6" />
          <OrangeBtn symbol="-" />
        </div>
        <div className="row">
          <GrayBtn symbol="1" />
          <GrayBtn symbol="2" />
          <GrayBtn symbol="3" />
          <OrangeBtn symbol="+" />
        </div>
        <div className="row">
          <BigGrayBtn symbol="0" />
          <GrayBtn symbol="." />
          <OrangeBtn symbol="=" />
        </div>
      </div>
    );
  }
}

export default Calculator;
