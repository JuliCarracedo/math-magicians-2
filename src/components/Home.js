import React from 'react';

const Home = () => (
  <div className="home">
    <h2>Welcome to our page</h2>

    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      Curabitur efficitur nisl vel leo congue, non aliquet urna malesuada.
      Mauris consectetur eget dolor sollicitudin vestibulum.
      Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae;
      Nulla facilisi. Phasellus et feugiat diam.
      Aenean id lorem accumsan, auctor nisl eget, congue lectus.
      Nullam congue vel nisi vitae interdum.
      Integer lectus nibh, tincidunt eget libero sed, sagittis maximus quam.
      Cras lobortis vestibulum dolor, vel feugiat leo convallis eget.
    </p>
  </div>
);

export default Home;
