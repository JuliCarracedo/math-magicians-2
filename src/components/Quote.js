import React from 'react';

const Quote = () => (
  <div>
    <p>
      &quot;Mathematics is language that God used to write the world&quot;
      : Galileo Galilei
    </p>
    <p>
      &quot;Mathematics seems to give one a new meaning&quot;
      : Charles Darwin
    </p>
    <p>
      &quot;The essence of mathematics is not to make simple things complicated,
      but make complicated things simple&quot;
      :S. Gudder.
    </p>
    <p>
      &quot;Pure mathematics is, in its form, the poetry of logical ideas&quot;
      : Albert Einstein
    </p>
  </div>
);

export default Quote;
