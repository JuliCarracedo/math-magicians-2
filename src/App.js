import './App.css';
import React from 'react';
import { BrowserRouter as Router, Routes, Switch } from 'react-router-dom';
import Navbar from './components/Navbar';
import Calculator from './components/Calculator';
import Home from './components/Home';

function App() {
  return (
    <React.StrictMode>
      <Router>
        <Navbar />
        <Routes>
          <Route exact path="/" render={() => <Home />} />
          <Route path="/calc" render={() => <Calculator />} />
        </Routes>
      </Router>
    </React.StrictMode>
  );
}

export default App;
