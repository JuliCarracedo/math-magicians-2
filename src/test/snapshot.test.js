
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import Calculator from '../components/Calculator';
import Home from '../components/Home';
import Quote from '../components/Quote';

it('renders correctly', () => {
  const tree = renderer
    .create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Quote />).toJSON();
  expect(tree).toMatchSnapshot();
});
