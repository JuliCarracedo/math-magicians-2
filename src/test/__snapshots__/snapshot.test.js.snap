// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    className="navbar"
  >
    <h2>
       MATH MAGICIANS
    </h2>
    <div
      className="links"
    >
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
         Home 
      </a>
      <a
        className="link"
        href="/calc"
        onClick={[Function]}
      >
         Calculator 
      </a>
      <a
        className="link"
        href="/quote"
        onClick={[Function]}
      >
         Quote 
      </a>
    </div>
  </nav>,
  <div
    className="home"
  >
    <h2>
      Welcome to our page
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur efficitur nisl vel leo congue, non aliquet urna malesuada. Mauris consectetur eget dolor sollicitudin vestibulum. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nulla facilisi. Phasellus et feugiat diam. Aenean id lorem accumsan, auctor nisl eget, congue lectus. Nullam congue vel nisi vitae interdum. Integer lectus nibh, tincidunt eget libero sed, sagittis maximus quam. Cras lobortis vestibulum dolor, vel feugiat leo convallis eget.
    </p>
  </div>,
]
`;

exports[`renders correctly 2`] = `
<div
  className="calc-section"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="calc-body"
  >
    <div
      className="screen"
    >
      <p
        className="output"
      >
        0
      </p>
    </div>
    <div
      className="row"
    >
      <button
        className="gray-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          AC
        </p>
      </button>
      <button
        className="gray-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          +/-
        </p>
      </button>
      <button
        className="gray-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          %
        </p>
      </button>
      <button
        className="orange-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          รท
        </p>
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="gray-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          7
        </p>
      </button>
      <button
        className="gray-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          8
        </p>
      </button>
      <button
        className="gray-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          9
        </p>
      </button>
      <button
        className="orange-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          x
        </p>
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="gray-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          4
        </p>
      </button>
      <button
        className="gray-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          5
        </p>
      </button>
      <button
        className="gray-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          6
        </p>
      </button>
      <button
        className="orange-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          -
        </p>
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="gray-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          1
        </p>
      </button>
      <button
        className="gray-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          2
        </p>
      </button>
      <button
        className="gray-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          3
        </p>
      </button>
      <button
        className="orange-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          +
        </p>
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="big-gray-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          0
        </p>
      </button>
      <button
        className="gray-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          .
        </p>
      </button>
      <button
        className="orange-btn"
        onClick={[Function]}
        type="button"
      >
        <p>
          =
        </p>
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly 3`] = `
<div
  className="home"
>
  <h2>
    Welcome to our page
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur efficitur nisl vel leo congue, non aliquet urna malesuada. Mauris consectetur eget dolor sollicitudin vestibulum. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nulla facilisi. Phasellus et feugiat diam. Aenean id lorem accumsan, auctor nisl eget, congue lectus. Nullam congue vel nisi vitae interdum. Integer lectus nibh, tincidunt eget libero sed, sagittis maximus quam. Cras lobortis vestibulum dolor, vel feugiat leo convallis eget.
  </p>
</div>
`;

exports[`renders correctly 4`] = `
<div
  className="quote"
>
  <p>
    "Mathematics is language that God used to write the world" : Galileo Galilei
  </p>
  <p>
    "Mathematics seems to give one a new meaning" : Charles Darwin
  </p>
  <p>
    "The essence of mathematics is not to make simple things complicated, but make complicated things simple" :S. Gudder.
  </p>
  <p>
    "Pure mathematics is, in its form, the poetry of logical ideas" : Albert Einstein
  </p>
</div>
`;
